CanvasPlusPlayground/Features/Files/FoldersPageView.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  FoldersPageView.swift
+//  CanvasPlusPlayground
+//
+//  Created by Abdulaziz Albahar on 12/10/24.
+//
+
+import SwiftUI
+
+struct FoldersPageView: View {
+    let course: Course
+    @State var folder: Folder?
+    @State private var filesVM: CourseFileViewModel
+
+    init(course: Course, folder: Folder? = nil, prevFilesVM: CourseFileViewModel? = nil) {
```

Comment:
```suggestion
    init(course: Course, folder: Folder? = nil, tranversedFolderIDs:[String]? = nil) {
```

I don't think we should pass view models and create new ones like this. Does this solution work better? You can set the traversed id's like so:

```swift
    _filesVM.traversedFolderIDs = traversedFolderIDs ?? []
```